web/source/keymanweb.js

```diff
@@ -2476,43 +2452,13 @@ if(!window['keyman']['initialized']) {
 
       var keyboardID = keymanweb._ActiveKeyboard == null ? '' : keymanweb._ActiveKeyboard['KI'];
 
-      if(keymanweb._LastActiveElement._kmwAttachment.keyboard != null) {
-        if(!keymanweb._JustActivatedKeymanWebUI) {
-          keymanweb.setActiveKeyboard(keymanweb._LastActiveElement._kmwAttachment.keyboard,
-            keymanweb._LastActiveElement._kmwAttachment.languageCode); 
-        } else {
-          keymanweb._LastActiveElement._kmwAttachment.keyboard = keyboardID;
-          keymanweb._LastActiveElement._kmwAttachment.languageCode = keymanweb.getActiveLanguage();
-        }
-      } else if(priorElement) { // Avoids complications with initialization and recent detachments.
-        if(!keymanweb._JustActivatedKeymanWebUI) {
-          keymanweb.setActiveKeyboard(keymanweb.globalKeyboard, keymanweb.globalLanguageCode); 
-        } else {
-          keymanweb.globalKeyboard = keyboardID;
-          keymanweb.globalLanguageCode = keymanweb.getActiveLanguage();
-        }
-      }
-
-      //TODO: the logic of the following line doesn't look right!!  Both variables are true, but that doesn't make sense!
-      //_Debug(keymanweb._IsIEEditableIframe(Ltarg,1) + '...' +keymanweb._IsMozillaEditableIframe(Ltarg,1));
-      if(!keymanweb._IsIEEditableIframe(Ltarg,1) || !keymanweb._IsMozillaEditableIframe(Ltarg,1))
-      {
-        keymanweb._DisableInput = 1; 
-        return true;
-      }
-      keymanweb._DisableInput = 0;
-
-      if(!keymanweb._JustActivatedKeymanWebUI)
-      {
-        keymanweb._DeadKeys = [];
-        keymanweb._NotifyKeyboard(0,Ltarg,1);  // I2187
+      if(keymanweb._JustActivatedKeymanWebUI) {
+        keymanweb._BlurKeyboardSettings();
+      } else {
+        keymanweb._FocusKeyboardSettings(priorElement ? false : true);
       }
-    
-      if(!keymanweb._JustActivatedKeymanWebUI  &&  keymanweb._SelectionControl != Ltarg)
-        keymanweb._IsActivatingKeymanWebUI = 0;
-      keymanweb._JustActivatedKeymanWebUI = 0;
 
-      keymanweb._SelectionControl = Ltarg;
+      keymanweb._CommonFocusHelper(Ltarg);
```

Comment:
This has a slightly different control flow for IE and Mozilla editable iframes. In the old version control would return at line #2501, see above.

Now, we ignore that return value in `_CommonFocusHelper` and continue and reset selection, etc, for iframes. I don't know if the new behaviour is correct, but it is certainly different.

Can you confirm which is the correct behaviour, and if the original behaviour is correct, fix the control flow for those cases?

Reply:
Oh.  That's an honest mistake on my part; didn't note the need to forward the return logic when I refactored the code.  I'll upload a fix correcting this to have the original logic's effect shortly.